### PR TITLE
M3-5177: Use React Query for Account Settings

### PR DESCRIPTION
**Pull request by bnussman**
## Description

- Remove `/v4/account/settings` data from Redux
- Uses **React Query**'s  _Query_ and _Mutation_ for `/v4/account/settings` 

## How to test

- Test all general use of Cloud Manager. Things that are most likely to break include `backups`, `object storage`, `managed`, `longview` and `linode network helper`.
- Ensure all everything at [`/account/settings`](http://localhost:3000/account/settings) still work as expected

### CODE REVIEW

File: packages/manager/src/features/Account/EnableManaged.tsx [hunk 1]
```diff
@@ -88,8 +87,8 @@ export const EnableManaged: React.FC<CombinedProps> = (props) => {
     enableManaged()
       .then(() => {
         handleClose();
-        update({ managed: true });
-        //push('/managed');
+        updateAccountSettingsData({ managed: true });
+        // push('/managed');
```
Comment (codyfinn) [inline packages/manager/src/features/Account/EnableManaged.tsx:91]: Can you remove this comment if it is not used?

**Review by codyfinn — APPROVED** [review]
I left a nitpicking comment but other than that this looks great! Nice job moving us closer to complete react query usage!

File: packages/manager/src/store/backupDrawer/index.ts [hunk 2]
```diff
@@ -293,23 +296,20 @@ export const enableAutoEnroll: EnableAutoEnrollThunk = () => (
...
   const state = getState();
   const { backups } = state;
-  const hasBackupsEnabled = pathOr(
-    false,
-    ['__resources', 'accountSettings', 'data', 'backups_enabled'],
-    state
-  );
   const shouldEnableBackups = Boolean(backups.autoEnroll);
+  const { data: accountSettings } = useAccountSettings();
+  const { mutateAsync: updateAccountSettings } = useMutateAccountSettings();
```
Comment (bnussman) [inline packages/manager/src/store/backupDrawer/index.ts:301]: OH NO :0 I made an invalid hook call without even noticing this.

**Review by ndukatz — APPROVED** [review]
Looks good! all E2E passed. all the areas you specified looked good in manual checks as well. Also tested managed.

File: packages/manager/src/features/Account/GlobalSettings.tsx [hunk 3]
```diff
@@ -103,38 +81,53 @@ const GlobalSettings: React.FC<CombinedProps> = (props) => {
...
+
+  const {
+    backups_enabled,
+    managed,
+    network_helper,
+    object_storage,
+  } = accountSettings;
+
+  const toggleAutomaticBackups = () => {
+    updateAccount({ backups_enabled: !backups_enabled }).catch(displayError);
```
Comment (HanaXu) [inline packages/manager/src/features/Account/GlobalSettings.tsx:107]: Do we want to return here as well?

**Review by HanaXu — APPROVED** [review]
LGTM!

File: packages/manager/src/features/Account/GlobalSettings.tsx (hunk 3 above)
Comment (bnussman) [reply packages/manager/src/features/Account/GlobalSettings.tsx:107]: Good catch.  Going to remove the returns, they are not needed.

**Pull request merged**